Build Linux aarch64 packages using the new GH runner

hexchain opened the pull request:
This PR tries to build one more artifact using the new GH ARM64 Linux runner (https://github.blog/changelog/2025-01-16-linux-arm64-hosted-runners-now-available-for-free-in-public-repositories-public-preview/).

I maintain the Insomnia flatpak on FlatHub (https://github.com/flathub/rest.insomnia.Insomnia) and building the application from source has been difficult. The environment for the compile/build phase is fully offline, which means any dependencies must be obtained beforehand, which in turn requires declaring them in the flatpak manifest. It is impossible to do this by hand for a large project like this. There is [a tool](https://github.com/flatpak/flatpak-builder-tools/tree/master/node) to help generate this , but it doesn't support git sources, which this application depends on.

I've stumbled upon the news that GHA added ARM64 Linux runners and decided to give this a try. If it works and it's acceptable, hopefully we can rely on the official aarch64 linux binaries in the future.

hexchain committed:
chore: add arch into AppImage name template

hexchain committed:
chore: run release workflows on aarch64 linux runner

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Kong/insomnia?pullRequest=8306) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Kong/insomnia?pullRequest=8306) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Kong/insomnia?pullRequest=8306) it.</sub>

jackkav commented:
Neat, we'll try to help

jackkav commented:
Update on this: recent changes to github runner environments across all OS appear to causing all the exception tests to fail on any changes to node-libcurl.

Going forward I can temporarily comment out these tests, just to see if this PR works, but ideally we would want to re-enable them before completing this work.

hexchain commented:
Those 4 tests also fail locally for me, on both x86_64 and aarch64, regardless of whether the binary addon is prebuilt or not.

filfreire commented:
@jackkav @hexchain we can retry with [node-libcurl 2.33.7](https://github.com/Kong/node-libcurl/releases/tag/v2.33.7)

hexchain commented:
Will update after that version become available on NPM.

Out of interest, why is the node-libcurl a fork? The original one also seems active at the moment. If there are non-upstreamable changes, is there any plan to rebase?

hexchain committed:
chore: upgrade @getinsomnia/node-libcurl to 2.33.7

filfreire reviewed (commented):
If I'm not mistaken, these also need to be adapted to this change, at least pointing to x64 build for the time being

- https://github.com/Kong/insomnia/blob/develop/.github/workflows/release-publish.yml#L250
- https://github.com/Kong/insomnia/blob/develop/.github/workflows/release-publish.yml#L270

hexchain committed:
fix: also consider process.arch when determining insomnia binary path

hexchain committed:
fix: lift unprivileged userns restrictions for linux

Ubuntu 24.04 introduced this new restriction so we just lift it for all
Linux runners now to avoid confusion in the future.

jackkav commented:
>Out of interest, why is the node-libcurl dependency a fork? The original one also seems active at the moment. If there are non-upstreamable changes, is there any plan to rebase?

https://github.com/JCMais/node-libcurl/pull/411#issuecomment-2562876841

I don't believe there are significant non-upstreamable changes. 

Our motivations have generally been keeping our electron/nodejs up to date, which releases every 3 months.

We are eagerly anticipating the N-API work in progress in the upstream to be completed, in order to upstream our changes and return to it. Then it will no longer be necessary for us to publish a new node-libcurl version for every nodejs upgrade. This will be a huge relief as it cut out several days of work every 2-3 months preparing node-libcurl releases.

hexchain commented:
The CI should pass for aarch64 now. I'll look at the release-publish workflow later.

hexchain committed:
chore: remove hardcoded platform in inso Dockerfile

hexchain committed:
chore: add architecture id into inso artifact name

hexchain committed:
feat: modify workflows to support multi-arch

hexchain commented:
I've updated the release-* workflows but cannot test if it works (and I'm not too confident about it)...

jackkav commented:
Future work here is running a beta release and checking all the linux dist channels for the arm build. Snap craft and such.
